distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Public_Utils.enso

```
@@ -24,7 +24,7 @@ get_required_field (key : Text) js_object (show_value : Boolean = True) (expecte
             suffix = if show_value then " in "+js_object.to_display_text else ""
             Error.throw (Enso_Cloud_Error.Invalid_Response_Payload "Missing required field `"+key+"`"+suffix+".")
         result = js_object.get key if_missing=handle_missing
-        if result.is_a expected_type then result else
+        result.if_not_error <| if result.is_a expected_type then result else
```

Why is this needed? 
Shouldn't `result.is_a` rethrow the error?

---

No, `is_a` does not propagate:
```
> (Error.throw "A")
>>> (Error: 'A')
> (Error.throw "A") . is_a Any
>>> true
> (Error.throw "A") . is_a Error
>>> true
> (Error.throw "A") . is_a Text
>>> false
> (Error.throw "A") . is_a Integer
>>> false
```

as you can see `is_a` 'swallows' the error and will return `true` for `x.is_a Error`.
We can change that behaviour, but until then, this `if_not_error` is necessary.